packages/core/src/Site/index.ts

```diff
@@ -690,6 +687,7 @@ export class Site {
       await this.writeSiteData();
       this.calculateBuildTimeForGenerate(startTime, lazyWebsiteGenerationString);
       if (this.backgroundBuildMode) {
+        // @ts-ignore
```

Comment:
could you explain why this is necessary?

Reply:
From what I understand, `_backgroundBuildNotViewedFiles` doesn't take in any args and isn't compatible with the `func` param in `delay` technically (not sure why `func` in `delay` is given its type `(arg: T[]) => Promise<unknown>` in the first place, could have been more flexible?). So TS gave me an error as it thought `backgroundBuildNotViewedFiles` should be receiving an arg when it didn't have to.

It was a little confusing here as TS has accepted `_backgroundBuildNotViewedFiles` as its `func` param in the first place, not giving any error. Not really sure why..

Reply:
Do you think it will be better if we assert the function signature as without argument at line 1706:
```
  backgroundBuildNotViewedFiles = delay(this._backgroundBuildNotViewedFiles as () => Bluebird<unknown>, 1000);
```
to make it more explicit that we know this function takes in no extra argument? https://www.scaler.com/topics/typescript/typescript-as/

Reply:
Thanks for the tip @tlylt 

On a related note, currently in `serveUtil.js`, in line 29 and 62, we have:

`return site.rebuildSourceFiles(filePath);`

even though rebuildSourceFiles does not take in any argument. Is this is mistake and should I rectify it?